domain/access/state/permission.go

```diff
@@ -561,10 +560,11 @@ func targetExists(ctx context.Context, tx *sqlair.TX, target corepermission.ID)
 	var targetExists string
 	switch target.ObjectType {
 	case coredatabase.ControllerNS:
-		if target.Key != coredatabase.ControllerNS {
-			return fmt.Errorf("%q %w", target, accesserrors.PermissionTargetInvalid)
-		}
-		return nil
+		targetExists = `
+SELECT  uuid AS &M.found
+FROM    controller
+WHERE   uuid = $M.grant_on
```

Comment:
Please be consistent with the other cases of the switch. `controller.uuid` rather than `uuid`, or swap the others.

Reply:
Good point, changed to be consistent